packages/test/test-end-to-end-tests/src/test/sharedTreeEndToEndTests.spec.ts

```diff
@@ -0,0 +1,461 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { describeCompat } from "@fluid-private/test-version-utils";
+import type { IContainer } from "@fluidframework/container-definitions/internal";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { ITestObjectProvider } from "@fluidframework/test-utils/internal";
+import {
+	ITree,
+	SchemaFactory,
+	TreeViewConfiguration,
+	type TreeView,
+} from "@fluidframework/tree";
+import {
+	asTreeViewAlpha,
+	SharedTree,
+	type Revertible,
+	type TreeViewAlpha,
+} from "@fluidframework/tree/internal";
+
+describeCompat("SharedTree", "NoCompat", (getTestObjectProvider, apis) => {
+	const { DataObject, DataObjectFactory } = apis.dataRuntime;
+	const { ContainerRuntimeFactoryWithDefaultDataStore } = apis.containerRuntime;
+
+	// An extension of Aqueduct's DataObject that creates a SharedTree during initialization and exposes it.
+	class DataObjectWithTree extends DataObject {
```

Comment:
This needlessly created a shared directory (as all DataObjects do). You could avoid an extra unrelated DDS using test utils if https://github.com/microsoft/FluidFramework/pull/23949 was merged (I don't think thats the right approach, but another example of a pattern which adds extra ddses and how to fix it).

https://github.com/microsoft/FluidFramework/pull/23943 Adds what this should probably use, so again something that doesn't exist yet.

Given the current state of things on main, and that this is called "DataObjectWithTree" so its clearly testing a data object that has a tree, (not is a tree) its probably fine as is, but I wanted to point out that this is involving SharedDirectory when it could be avoided.